packages/binarytree/src/binaryTree.ts

```diff
@@ -0,0 +1,750 @@
+import {
+  Lock,
+  bitsToBytes,
+  bytesToBits,
+  bytesToHex,
+  concatBytes,
+  equalsBits,
+  equalsBytes,
+  matchingBitsLength,
+  setLengthRight,
+} from '@ethereumjs/util'
+import debug from 'debug'
+
+import { CheckpointDB } from './db/index.js'
+import { InternalBinaryNode } from './node/internalNode.js'
+import { StemBinaryNode } from './node/stemNode.js'
+import { decodeBinaryNode, isInternalBinaryNode, isStemBinaryNode } from './node/util.js'
+import { type BinaryTreeOpts, ROOT_DB_KEY } from './types.js'
+
+import type { BinaryNode } from './node/types.js'
+import type { PutBatch } from '@ethereumjs/util'
+import type { Debugger } from 'debug'
+
+interface Path {
+  node: BinaryNode | null
+  remaining: number[]
+  stack: Array<[BinaryNode, number[]]>
+}
+
+/**
+ * The basic binary tree interface, use with `import { BinaryTree } from '@ethereumjs/binarytree'`.
+ */
+export class BinaryTree {
+  /** The options for instantiating the binary tree */
+  protected _opts: BinaryTreeOpts
+
+  /** The root for an empty tree */
+  EMPTY_TREE_ROOT: Uint8Array
+
+  protected _db!: CheckpointDB
+  protected _hashLen: number
+  protected _lock = new Lock()
+  protected _root: Uint8Array
+
+  protected DEBUG: boolean
+  protected _debug: Debugger = debug('binarytree:#')
```

Comment:
why the "#" on the end here?

Reply:
This is the new logging convention we use for the base topic for any package (so `mpt:#`, `evm:#`, etc).  There was some screwiness with doing `DEBUG=ethjs,evm*` as opposed to `DEBUG=ethjs,evm:*` where subtopics weren't getting logged or something.